clion/src/main/java/org/sonarlint/intellij/clion/CFamilyFileExclusionContributor.java

```diff
@@ -0,0 +1,20 @@
+package org.sonarlint.intellij.clion;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.sonarlint.intellij.common.analysis.ExcludeResult;
+import org.sonarlint.intellij.common.analysis.FileExclusionContributor;
+import org.sonarlint.intellij.common.ui.SonarLintConsole;
+
+public class CFamilyFileExclusionContributor implements FileExclusionContributor {
+
+  @Override
+  public ExcludeResult shouldExclude(Module module, VirtualFile fileToAnalyze) {
+    AnalyzerConfiguration.ConfigurationResult configurationResult = new AnalyzerConfiguration(module.getProject()).getConfiguration(fileToAnalyze);
+    if (configurationResult.hasConfiguration()) {
+      return ExcludeResult.notExcluded();
+    }
+    SonarLintConsole.get(module.getProject()).debug(configurationResult.getSkipReason());
```

Comment:
If I'm not mistaken, there is already a debug log in the calling code

Reply:
@henryju ideed, I checked and it is the case, nice catch.